File: APP(Android)/lib/domain/goal/goal.dart
class Goal {
  final int? id;
  final String? goal;
  final int? objectiveWon;
  final int? targetId;

  Goal.fromJson(Map<String, dynamic> json)
      : id = json["id"],
        goal = json["goal"],
        objectiveWon = json["objective_won"],
        targetId = json["targetId"];
}

File: APP(BE)/server/src/entity/goal.ts
import {
  Entity,
  PrimaryGeneratedColumn,
  Column,
  ManyToOne,
  BaseEntity,
  JoinColumn,
} from "typeorm";
import { Target } from "./target";

@Entity()
export class Goal extends BaseEntity {
  @PrimaryGeneratedColumn()
  id: number;

  @Column({ length: 100 })
  goal: string;

  @Column({ type: "int" })
  objective_won: number;

  @Column({ type: "int" })
  targetId: number;

  @ManyToOne(() => Target, (target) => target.donations)
  @JoinColumn({ name: "targetId" })
  target: Target;
}
